### DISCUSSION TITLE
meta_hours_3

### DISCUSSION THREAD

**Discussion by yangwao** [kodadot/nft-gallery#1710] [Meta-Hours]
![image](https://user-images.githubusercontent.com/5887929/150183415-9c9608a8-5cca-4b61-8c59-b2da2a09d12b.png)

![stairs](https://user-images.githubusercontent.com/5887929/149831026-8d5b287d-493c-4cb5-9159-eafe2c55c4b7.jpeg)

[Discord event](https://discord.gg/Z5M4D6BGKQ?event=928948265204990013)

### Content is being prepared

Wondering what will be done and what else will come! 

You can peak on [past episode Meta_hours_2 till then](https://github.com/kodadot/nft-gallery/discussions/1699)
(this placeholder will be edited further)

## Brief numbers 
- our contributions are on a steady rise, which is good I guess, we've got a load. 
- Briefly merged PRs are accumulating and we are working on ways how to add more recurring contributors, will talk later about that  

![image](https://user-images.githubusercontent.com/5887929/150175989-b7529c4c-cbf2-4be4-84ec-9baec7c267c8.png)

Some numbers as we've been switching from Netlify to Cloudflare, you won't see Netlify dashboard anymore 😄

Speaking from the cadence of development, we've got like a new all-time high number of daily builds.
Each build is made once pull requests is open and commit is made to opened PR so there is a new build every time feedback loop happens and what's needs to be done for better code quality or feature to be ready for shipping to the production

![image](https://user-images.githubusercontent.com/5887929/149328189-4349edaa-1c88-4cc3-824a-f3ca31dc4e57.png)

We've used a significant amount of build minutes, but this number will be of the past as we are slowly switching to CloudFlare where they don't count minutes.

![image](https://user-images.githubusercontent.com/5887929/149328168-a33a6d88-5139-4dc8-828a-d6a8ebac1503.png)

Bandwidth between bill period 21/12-21/1 -- taken on 13/1 so there is approx room for growth for crossing usual 1TB for static assets like js,CSS,img is not bad. NFTs are not included in this number.

![image](https://user-images.githubusercontent.com/5887929/149328315-c1b2ccf3-ecae-4ff1-ae91-a7a2dc124602.png)

That falling tail is as we are switching to Cloudflare so we will be bringing new data and will have more insightful peaks on users behaviour as we have are combining web analytics from CloudFlare which protects user privacy and Google Analytics which does not.

![Screenshot 2022-01-13 at 16 45 58](https://user-images.githubusercontent.com/5887929/149362272-f4c4d209-4aa5-4da4-afd5-0312d71d0305.png)
![Screenshot 2022-01-13 at 16 46 04](https://user-images.githubusercontent.com/5887929/149362278-36b6899c-0dda-49fc-a5b3-42eec2ba7e5c.png)

### War-room -- all hands-dev chill session
- On 14/1/2022 we've had a first unofficial war-room session. War-room is our discord channel for recurring contributors to the KodaDot codebase. 
- We've chilled discussion about the recent Beta release and first-time "voice-meeting" between developers about it. 
- It has really magical energy to exchange thoughts on development, reflections and learnings.
- Best of it, took over one hour. 
 
<img width="534" alt="Screenshot 2022-01-14 at 17 42 43" src="https://user-images.githubusercontent.com/5887929/149552430-381c9b58-ebb1-415d-a548-5971db18bfbc.png">

### so we've released Beta 
- it did not go ... without issues... And that's was great learning for us.
- it was 10 minutes before usual dinner time. 
 
<img width="510" alt="Screenshot 2022-01-12 at 16 40 48" src="https://user-images.githubusercontent.com/5887929/149343710-b59bd1e4-b62e-4b75-bf43-dab8876dfc93.png">

### Thing with service-worker.js
Thing was that we had an issue with `service-worker.js` from the old instance. 
Briefly explain that a service worker is a small autonomous agent which caches your data on the client-side so even you are **without the internet**, you can keep browsing assets on your client. Thing was that new technology we've been rewriting whole KodaDot -- Nuxt, was serving service-worker.js differently, nothing big once we figured out stuff, it was on sw.js.. but the thing was in our setup Nuxt was only building `sw.js` and when your old client was still fetching **https://kodadot.xyz/service-worker.js it was not there**, so **your client did not fetch new assets at all.**
 
- Once we had figured a way that we need **"simulate" having old `service-worker` we had to push "new" revision**.. and voila that was it.. yeah it wasn't gone easy as Safari just pop toast and that's it. Firefox/Chrome just automatically when you browse within the app.. fetched new content so I've personally definitely learned a bit about the history of service workers in the past 5 years and we have few ideas how we can leverage KodaDot being more **offline-first app**

- What had made the **whole "hacking" and "guessing" degree harder was that builds behaved really differently on localhost vs on builds made by CDN**, plus Cloudflare Pages did not serve `nft.kodadot.xyz` and whole "cache" issues and plus inspect network requests for a service worker is really a trial error or like looking needle in a haystack.

https://github.com/kodadot/nft-gallery/issues/1797

### Security 

- Hence we have a new deployment on Cloudflare, we've forgotten to set `_headers` and now they are back, so celebrating security https://github.com/kodadot/nft-gallery/issues/1894 🥳
![image](https://user-images.githubusercontent.com/5887929/150113898-6a4f9572-c937-4e8b-9a21-ba020a86f74f.png)

### Minting broken
- We've got the report back by @jkdizajn that our functions aren't working, well, `window.origin` was for https://kodadot.xyz which was actually deployed on the Cloudflare but not on the pointing to Netlify anymore, we've introduced quick fix by @vikiival which.. did not worked out because CORS stuff. 
- To nail stuff down, @vikiival has implemented Estuary.tech in this PR and it's working now from production version https://github.com/kodadot/nft-gallery/pull/1931

### Repository cleaning
- Fuckup with renaming repository from `main-nuxt` to `main` 
-- This one was really funny. As we've had code in `main` where was our old SPA app, we tried to do cleaning but git was not fan of it and it go really bad ass and few integrations did not survive as no one is expecting renaming branch to one which was existing before, duh
-- Some contributors had issues as they did not sync upstream and apparently upstream sync doesn't sync branch renaming at all and once you are not aware of old branch should be gone and rename new to old name, it go really black
- Takeaway, don't rename branches probably

- We've introduced formatted code and we have now Prettier, which should bring more clean code and more consistency for contributors https://github.com/kodadot/nft-gallery/pull/1862

- We've polished README.md for new contributors so we are putting emphasis how to spin up KodaDot much faster and contributor easier
- https://twitter.com/KodaDot/status/1482706769151512579

- ![image](https://user-images.githubusercontent.com/5887929/149675766-be799a5f-9b18-40e7-b24b-857475800ed9.png)

### Bringing new developers on board
- We've got offer from US colleagues to have their interns join our team, it's something experimental and we will see how it goes
- We plan to put few of our bigger issues on Gitcoin so we can attract new steady recurring developers to KodaDot

### Cherry on top
- Hence we've been rolled to CloudFlare Pages Beta channel, somehow we break their building stuff and stopped building our site for two days tho. 
- Good news are, we are back and building our site well!

### To be told by others 
- Package & Maintainer Health done by @kkukelka  -- https://github.com/kodadot/nft-gallery/discussions/1851
- @vikiival -- https://github.com/kodadot/nft-gallery/issues/1889 
- https://github.com/kodadot/nft-gallery/pull/1925 -- we can store big files up to 32G ? https://github.com/kodadot/nft-gallery/issues/963
- @kkukelka wrote article to summarize our quartal progress for '21Q4 and what was done -- https://medium.com/kodadot/road-to-kodadot-v2-recapping-the-last-chapter-of-2021-9709fb4f3ee7

**Comment by metakeller** [comment]
Hey! As you might know we are launching our second collection, KuBots

We plan to give away a couple of Eggs to the greatest communities on the market. What do you think about hosting this type of giveaway on your page? We are ready to give away 2 eggs for KodaDot community. :)

**Reply by yangwao** [reply]
> Hey! As you might know we are launching our second collection, KuBots
> 
> We plan to give away a couple of Eggs to the greatest communities on the market. What do you think about hosting this type of giveaway on your page? We are ready to give away 2 eggs for KodaDot community. :)

Yes, we can announce it for the very last, wenmintsir will be around hopefully! 😄

**Comment by vikiival** [comment]
**MetaHours** by @vikiival

Cloudflare Workers:
**durable-jpeg** 
- People put their high res Jpegs into the KodaDot and sometimes an image can have like 13 MB if you multiply it by 12 which is the number on images you can get 150 MB per page, 
- therefore we decided to leverage Cloudflare Images - the service which can store and transform images on the go using like 250 content delivery networks around the planet. 
- So first hit we downloaded all the images from the KodaDot - they have 113 GBs (wow :O) 
- then we uploaded every image to Cloudflare and the image id we saved to the key-value storage
- What does it look like on the front end? instead of loading big images, you will get images in resolution 600 by 600 px 

**pinning**
 Service similar to Pinata API, but uses Estuary - the good thing is that you can pin files to almost 32 GBs
As a bonus, it's written in Rust

**direct-upload**
Each image minted via KodaDot is automatically stored on the Cloudflare images to provide the better user experience


KodaDot:
- Replacing Pinata with Estuary #1925 
- Collection refactor #1934
- Changed background image to Webp #1831
- Added JEST #1802 
- Fixed VR branch for @rchovatiya88 

by Leo (@roiLeo ):
- A lot of contributions to the rubick - the indexer by SubSquid, backend sorting
- Stuff to make developer experience better
- CookieBanner background & border

by Darshana (@prachi00 ):
- #1814 History component refactors 
- #1887 Hide disabled pagination buttons 
- #1421 Add storing Discord to Identity storage 

[**Bao**](https://github.com/kodadot/bao)
- a repository with a lot of useful hacks

**Comment by damskyftw** [comment]
Metahours by damsky

- post-covid work life balance - getting back to the shape
- updated https://github.com/kodadot/nft-gallery/issues/1604#issuecomment-1015270618
- finished k-overview on discord
- creating materials for #1445 and having first 2 candidates, which will be fully testing the whole product without any hints and posting their feedback :)
- will be updating the #1506 tutorial so everyone can work on it even without coding skills with rewards - #1891 
- #1808

**Reply by yangwao** [reply]
we should invite some cosplay waifus 👀

**Comment by yangwao** [comment]
HD recording incoming soon https://youtu.be/F4W7S4UbbXw

**Comment by yangwao** [comment]
Continue to [Meta_Hours_4](https://github.com/kodadot/nft-gallery/discussions/2007)
